nixos/modules/virtualisation/qubes-dom0.nix

```diff
@@ -0,0 +1,117 @@
+      pkgs.qubes-linux-utils
+    ];
+    systemd.tmpfiles.settings."10-qubes" = {
+      "/var/log/qubes".d = {
+        user = "qubes";
+        group = "qubes";
+        mode = "700";
+      };
+      # Used to recognize if current system is qubes dom0.
+      # https://github.com/QubesOS/qubes-core-admin-client/blob/d9db37fdae9aaf91dec993b5dd111e02e66f782d/qubesadmin/__init__.py#L31
+      "/etc/qubes-release".f = {};
+    };
+    systemd.packages = [
+      pkgs.qubes-core-qubesdb.daemon
+      pythonPackages.qubes-core-admin
+      pkgs.qubes-linux-utils
+    ];
+
+    # Relevant for domU, dom0 uses qubes-db-dom0
+    systemd.services.qubes-db.enable = false;
+    systemd.services.qubes-db-dom0 = {
+      wantedBy = ["multi-user.target"];
+      after = ["systemd-tmpfiles-setup.service"];
+    };
+    # FIXME: Wrap qubesd binary instead.
+    systemd.services.qubesd = {
+      wantedBy = ["multi-user.target"];
+      before = ["systemd-user-sessions.service"];
+      path = [pkgs.lvm2];
+    };
+    systemd.services.qubes-qmemman.wantedBy = ["multi-user.target"];
+    systemd.services.qubes-meminfo-writer.enable = false;
+    systemd.services.qubes-meminfo-writer-dom0 = {
+      wantedBy = ["multi-user.target"];
+      before = ["systemd-user-sessions.service"];
+      after = ["qubes-core.service" "qubes-qmemman.service"];
+    };
+    # TODO: It is just runs startup script, I believe it would be better to just replace the whole service.
+    systemd.services.qubes-core = {
+      wantedBy = ["multi-user.target"];
+      aliases = ["qubes_core.service"];
+    };
+    systemd.services."qubes-vm@" = {
+      wantedBy = ["multi-user.target"];
+      after = ["qubesd.service" "qubes-meminfo-writer-dom0.service"];
+    };
+    systemd.services."qubes-reload-firewall@".wantedBy = ["multi-user.target"];
+
+    virtualisation.xen = {
+      enable = true;
+      # TODO: Wait until qubes xen fork is merged.
+      # package = pkgs.qubes-vmm-xen;
+      # stored = "${pkgs.qubes-vmm-xen}/bin/xenstored";
+      # package-qemu = pkgs.qemu_qubes;
+
+      bootParams = [
+        "gnttab_max_frames=2048"
+        "gnttab_max_maptrack_frames=4096"
+      ] ++ optional (!cfg.smt) "smt=off";
+
+      # TODO: Add declarative xl.conf
+      # settings = {
+      #   autoballoon = "0";
+      #   lockfile = "/var/run/qubes/xl-lock";
+      # };
+    };
+
+    environment.systemPackages = [
+      # FIXME: qrexec-policy-exec system path is hardcoded in qubes-core-qrexec.util
+      pythonPackages.qubes-core-qrexec
+      # FIXME: qubes-core-admin-client system path is hardcoded in qubes-core-admin
+      pythonPackages.qubes-core-admin-client
+      # TODO: add option for common tools, those are entirely optional.
+      pkgs.qubes-manager
+      pkgs.qubes-core-qubesdb
+
+      # Those are not technically optional, but might be disabled for non-gui nixos?
+      # qvm-appmenus - expected in $PATH by some tools, TODO: patch/wrap packages instead?
```

Comment:
I think it makes sense to have all qvm-X commands in PATH by default. Using it will also allow to fix some of dependency cycles.

It would be nice if the system worked without them, but I don't think it is possible without potentially huge QubesOS refactoring.

Reply:
There is currently no dependency cycles, but some parts of the stack assume qvm-* commands are in PATH.
I don't think it is worth rewrapping everything, Its fine as long as there is no dependencies with hardcoded /run/current-system paths